File: backend/src/InstitutoVirtus.API/Models/AuthModels.cs
namespace InstitutoVirtus.API.Models;

public class LoginRequest
{
    public string Email { get; set; } = string.Empty;
    public string Senha { get; set; } = string.Empty;
}

public class RegisterRequest
{
    public string NomeCompleto { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Senha { get; set; } = string.Empty;
    public string Telefone { get; set; } = string.Empty;
    public DateTime DataNascimento { get; set; }
    public string TipoPessoa { get; set; } = "Responsavel";
}

public class ForgotPasswordRequest
{
    public string Email { get; set; } = string.Empty;
}

public class ResetPasswordRequest
{
    public string Token { get; set; } = string.Empty;
    public string NewPassword { get; set; } = string.Empty;
}

public class LoginResponse
{
    public string Token { get; set; } = string.Empty;
    public string RefreshToken { get; set; } = string.Empty;
    public UserDto User { get; set; } = new();
    public int ExpiresIn { get; set; }
}

public class UserDto
{
    public string Id { get; set; } = string.Empty;
    public string Nome { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;
    public string? PessoaId { get; set; }
    public bool Ativo { get; set; } = true;
}

public class RefreshTokenRequest
{
    public string Token { get; set; } = string.Empty;
}

File: frontend/src/app/core/auth/models/auth.models.ts
export interface User {
  id: string;
  nome: string;
  email: string;
  tipo: UserRole;
  pessoaId?: string;
  ativo: boolean;
}

export type UserRole =
  | 'Admin'
  | 'Coordenador'
  | 'Professor'
  | 'Responsavel'
  | 'Aluno';

export interface LoginRequest {
  email: string;
  senha: string;
}

export interface LoginResponse {
  token: string;
  refreshToken: string;
  user: User;
  expiresIn: number;
}

export interface Session {
  user: User | null;
  token: string | null;
  refreshToken: string | null;
  isAuthenticated: boolean;
}
